### ISSUE TITLE
edit previously run query

### ISSUE THREAD

**Issue by mah11** [pombase/website#634]
It would be handy to be able to pick a query in the results list and edit it to yield a new query. This would be most useful for phenotype queries, because of the options for single- vs. multi-allele genotypes and expression levels - would save re-entering the FYPO name or ID.

It might also prove useful for combined queries - change one of the parameters by adjusting an ontology ID or other option, instead of having to do extra steps to add a parameter and then create a new combination to include it. And may help with #463 ...

**Comment by ValWood** [comment]
If you edit a query  would   the old query still be there?

**Comment by kimrutherford** [comment]
> If you edit a query would the old query still be there?

I'd say no to this I think.

But maybe we "Edit ..." and "Duplicate and edit ..." options for existing queries?

**Comment by mah11** [comment]
>> If you edit a query would the old query still be there?
> 
> I'd say no to this I think.

I disagree. If you edit the query, you've effectively created a different query, so to me it makes sense for the history to show both of the queries you've run. (My goal in making this request is to simplify entering query parameters.)

> But maybe "Edit ..." and "Duplicate and edit ..." options for existing queries?

ugh, no, too damn complicated.

**Comment by ValWood** [comment]
Would it save a lot of time for users?
Especially if they need to know the ontology ID?

It takes me around 10 seconds to create a fairy complicated query (compared to Intermine where it takes me the best part of an afternoon).

Let's keep this on the back burner with the other tickets for query builder improvements (which will probably not happen until after Browser)

**Comment by kimrutherford** [comment]
It's not query editing but I thought it might be useful to extract and re-run parts of a query.  So now if you view the query details, the descriptions and the `OR`/`AND` in the expanded query view are now clickable.  Clicking will take you to the results for just that sub-query and will add it to the query history if it's not there already.

![image](https://github.com/user-attachments/assets/773bb0b4-b1e8-4070-909a-a86a9e52726c)

![image](https://github.com/user-attachments/assets/574246d8-94ad-4487-af74-32eae0e934fe)

![image](https://github.com/user-attachments/assets/7e548d5b-d28f-4362-8fff-c97160d3277e)

**Comment by ValWood** [comment]
useful!

**Comment by ValWood** [comment]
I had forgotten that you could do that! @PCarme FYI!

I don't think we need to make queries editable, for the reasons above , and, because if you have a biq query, you can  rerun a sub part of the query and combine it with something else....
So if we agree, I will close this one tomorrow.
